src/device-api/common.js

```diff
@@ -27,20 +28,13 @@ export async function doRestart(appId, force) {
 			const imageIds = _.map(app.services, 'imageId');
 			applicationManager.clearTargetVolatileForServices(imageIds);
 
-			const currentServices = app.services;
-			app.services = [];
-			return deviceState
-				.pausingApply(() =>
-					deviceState
-						.applyIntermediateTarget(currentState, { skipLock: true })
-						.then(function () {
-							app.services = currentServices;
-							return deviceState.applyIntermediateTarget(currentState, {
-								skipLock: true,
-							});
-						}),
-				)
-				.finally(() => deviceState.triggerApplyTarget());
+			return deviceState.pausingApply(async () => {
```

Comment:
I understand this is the fix for #1509 @20k-ultra. I'm wondering why it was being done using intermediate targets before, was it to serialize the container restarts? What about doing it that way was causing the failure? 

I don't understand too much about intermediate targets so that might be cause of my confusion. 

Regarding the code itself, using `Promise.each` instead of `for` might make it more consistent with the rest of the code, but I don't see other issues.

Reply:
Tbh, I couldn't understand it much and it was written so differently from how V2 endpoints that I didn't want to invest the time to learn it as I was just applying how V2 performs a restart which is much easier to follow.